### ISSUE TITLE
[java] A false positive about the rule RedundantFieldInitializer

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#4070]
**Affects PMD Version:6.47.0** 

**Rule:RedundantFieldInitializer**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.47.0/pmd_rules_java_performance.html#redundantfieldinitializer>

**Description:**
In the following code example, I found a FP. Line 3 is not a redundant initializer, but it reports a warning.

**Code Sample demonstrating the issue:**
```java
public class FP {
    private static final int BASE = 1;
    char c = 0 + BASE;  // should not report a warning in this line
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*
